SYCL/DeviceLib/imf_double2half.cpp

```diff
@@ -0,0 +1,60 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// RUN: %clangxx -fsycl -fno-builtin -fsycl-device-lib-jit-link %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// UNSUPPORTED: cuda || hip
+
+#include "imf_utils.hpp"
+
+extern "C" {
+_iml_half_internal __imf_double2half(double);
+}
+
+int main() {
+
+  sycl::queue device_queue(sycl::default_selector_v);
+  std::cout << "Running on "
+            << device_queue.get_device().get_info<sycl::info::device::name>()
+            << "\n";
+
+  if (!device_queue.get_device().has(sycl::aspect::fp64)) {
+    std::cout << "Test skipped on platform without fp64 support." << std::endl;
+    return 0;
+  }
+
+  {
+    std::initializer_list<uint64_t> input_vals = {
```

Comment:
Is this just a sanity check or detailed test?
The test is good as a sanity one.
For detailed test I would suggest to extend it by:
* all DP corner points: +/-zero, +/-max&min subnormals, +/-max&min normals, +/-infs, nans
* all HP expected output corner points: +/-zero, +/-max&min subnormals, +/-max&min normals
(some of them are already covered)
* + all possible exponent fields (with random mantissa)

Reply:
Hi, @akolesov-intel 
This test is a lit test which is similar to sanity test. And I also added +0/-0, +/-max & min subnormals, +/- max & min normals, +/-infs
Thanks very much.